kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/SchemaManagerTest.java

```diff
@@ -110,14 +112,60 @@ public void testTimestampPartitionSet() {
     Assert.assertEquals("The field name does not match the field name of time partition",
         testField.get(),
         ((StandardTableDefinition) tableInfo.getDefinition()).getTimePartitioning().getField());
+    Assert.assertNull("Partition expiration is not null",
+        ((StandardTableDefinition) tableInfo.getDefinition()).getTimePartitioning().getExpirationMs());
+  }
+
+  @Test
+  public void testPartitionExpirationSetWithoutFieldName() {
+    Optional<Long> testExpirationMs = Optional.of(86400000L);
+    SchemaManager schemaManager = new SchemaManager(mockSchemaRetriever, mockSchemaConverter,
+        mockBigQuery, false, false, false, Optional.empty(), Optional.empty(), Optional.empty(), testExpirationMs, Optional.empty());
+
+    when(mockSchemaConverter.convertSchema(mockKafkaSchema)).thenReturn(fakeBigQuerySchema);
+    when(mockKafkaSchema.doc()).thenReturn(testDoc);
+
+    TableInfo tableInfo = schemaManager
+        .constructTableInfo(tableId, fakeBigQuerySchema, testDoc);
+
+    Assert.assertEquals("Kafka doc does not match BigQuery table description",
+        testDoc, tableInfo.getDescription());
+    Assert.assertEquals("The partition expiration does not match the expiration in ms",
+        testExpirationMs.get(),
+        ((StandardTableDefinition) tableInfo.getDefinition()).getTimePartitioning().getExpirationMs());
+    Assert.assertNull("Timestamp partition field name is not null",
+        ((StandardTableDefinition) tableInfo.getDefinition()).getTimePartitioning().getField());
+  }
+
+  @Test
+  public void testPartitionExpirationSetWithFieldName() {
+    Optional<Long> testExpirationMs = Optional.of(86400000L);
+    Optional<String> testField = Optional.of("testField");
+    SchemaManager schemaManager = new SchemaManager(mockSchemaRetriever, mockSchemaConverter,
+        mockBigQuery, false, false, false, Optional.empty(), Optional.empty(), testField, testExpirationMs, Optional.empty());
+
+    when(mockSchemaConverter.convertSchema(mockKafkaSchema)).thenReturn(fakeBigQuerySchema);
+    when(mockKafkaSchema.doc()).thenReturn(testDoc);
+
+    TableInfo tableInfo = schemaManager
+        .constructTableInfo(tableId, fakeBigQuerySchema, testDoc);
+
+    Assert.assertEquals("Kafka doc does not match BigQuery table description",
+        testDoc, tableInfo.getDescription());
+    Assert.assertEquals("The partition expiration does not match the expiration in ms",
+        testExpirationMs.get(),
+        ((StandardTableDefinition) tableInfo.getDefinition()).getTimePartitioning().getExpirationMs());
```

Comment:
Same comment here RE: pulling out the table definition into a separate variable.